src/torchcodec/samplers/_implem.py

```diff
@@ -179,9 +184,24 @@ def to_framebatch(frames: list[Frame]) -> FrameBatch:
             data=data, pts_seconds=pts_seconds, duration_seconds=duration_seconds
         )
 
-    all_decoded_frames: list[Frame] = [
-        decoder.get_frame_at(index) for index in all_clips_indices
-    ]
+    all_clips_indices_sorted, argsort = zip(
+        *sorted((j, i) for (i, j) in enumerate(all_clips_indices))
+    )
+    previous_decoded_frame = None
+    all_decoded_frames = [None] * len(all_clips_indices)
+    for i, j in enumerate(argsort):
+        frame_index = all_clips_indices_sorted[i]
+        if (
+            previous_decoded_frame is not None  # then we know i > 0
+            and frame_index == all_clips_indices_sorted[i - 1]
+        ):
+            # Avoid decoding the same frame twice.
+            decoded_frame = previous_decoded_frame
```

Comment:
To be slightly safer w.r.t. future changes this should be

```py
decoded_frame = copy(previous_decoded_frame)
```

but we don't implement `__copy__` on Frame.

Note that a copy *still* happens within `to_framebatch`, so this is currently safe, but admittedly subject to an implementation detail that will change.

We can either:

- be OK with this since we'll re-implement it in C++ anyway
- implement `__copy__`.

LMK.

Reply:
I am OK with this as-is.

Reply:
OK, I'll add a comment in `to_framebatch()` so we don't accidentally mess it up.